src/pymmcore_plus/core/_mmcore_plus.py

```diff
@@ -1734,24 +1736,32 @@ def saveSystemConfiguration(self, filename: str) -> None:
         **Why Override?** To also save pixel size configurations.
         """
         super().saveSystemConfiguration(filename)
-        px_configs = self.getAvailablePixelSizeConfigs()
-        if not px_configs:
-            return
         # saveSystemConfiguration does not save the pixel size config so here
         # we add to the saved file also any pixel size config.
         with open(filename, "a") as f:
-            f.write("# PixelSize settings")
-            for px_config in px_configs:
-                data = self.getPixelSizeConfigData(px_config)
-                obj = data.dict()["Objective"]["Label"]
-                px_size = self.getPixelSizeUmByID(px_config)
-                px_affine = self.getPixelSizeAffineByID(px_config)
-                cfg = (
-                    f"\nConfigPixelSize,{px_config},Objective,Label,{obj}\n"
-                    f"PixelSize_um,{px_config},{px_size}\n"
-                    f"PixelSizeAffine,{px_config},{','.join(map(str, px_affine))}"
+            self._save_pixel_configurations(f)
+
+    def _save_pixel_configurations(self, f: TextIOBase) -> None:
+        px_configs = self.getAvailablePixelSizeConfigs()
+        if not px_configs:
+            return
+        cfg = ["# PixelSize settings"]
+        for px_config in px_configs:
+            data = self.getPixelSizeConfigData(px_config).dict()
+            for device, prop in data.items():
+                cfg.extend(
+                    f"ConfigPixelSize,{px_config},{device},{k},{v}"
+                    for k, v in prop.items()

```

Comment:
```suggestion
            cfg.extend(
                f"ConfigPixelSize,{px_config},{device},{prop},{val}"
                for device, prop, val in self.getPixelSizeConfigData(px_config)
```

the config object is already an iterable that gives you everything you need.  no need to cast to a dict and then iterate over `items()`